### PR TITLE
write custom toast matcher

### PR DESCRIPTION
**Pull request by marinovdeyan**
The ToastMatcher is supposed to be a class that implements TypeSafeMatcher<Root?>. Its purpose is to look through the window hierarchy of the android phone and search for the text of the toast you want to assert existence.

On my side, even though I see the toast displaying in testAlreadyUsedPhoneNumber() (SignUpScreenTest.kt) with the exact same text as the one in the onView call, the window layout params type is 1 (so the application window, and not the overlayed stuff where the Toast should be located).

In other words, root?.windowLayoutParams?.get()?.type should be equal to 2038 here and not 1.

According to the internet, this sometimes happens on the emulator and they recommend testing with a real device. I do not have a real device at hand right now. Could someone test this with their Android phone?

### CODE REVIEW

**Comment by marinovdeyan** [comment]
After thorough testing with different Android SDKs, it turns out that anything >= SDK 30 cannot use this trick to match Toasts anymore. I have tested on SDK 29 and it works great, but not on greater versions.

I will unfortunately be obligated to abandon this PR and we need to think if we want to switch to using Snackbars (that can be tested with normal Compose UI testing) or if we don't test the Toasts that we output.
